src/frame.js

```diff
@@ -0,0 +1,44 @@
+// this handles the rolls and adding them to each frame, it only knows about the one frame
+class Frame {
+  constructor() {
+    this.roll1 = 0;
+    this.roll2 = 0;
+    this.rollCount = 1;
+    this.mark = 'none';
+    this.bonus1 = [0];
+    this.bonus2 = [0];
+  }
```

Comment:
I'm thinking:
* I'm not sure why the bonus arrays need to start with a length of one. Why not empty?
* Do they need to be arrays? After all, they only ever contain 0, or 0 and another number.
* Or you could have just one array, that has 0, 1 or 2 bonuses in it.